abilitybot/index.js

```diff
@@ -5,13 +5,19 @@ let lastEnemyId = -1;
 let charType;
 
 export function configureBot(rg) {
-  rg.characterType = CharInfo.type[Math.round(Math.random() * 1000000) % 4];
+  rg.characterConfig = {
+    characterType: CharInfo.type[Math.round(Math.random() * 1000000) % 4]
+  };
 }
 
 export async function processTick(rg) {
 
   // The character type we request may not be the one we actually get
-  charType = CharInfo.type.indexOf(rg.characterType);
+  const characterType = rg.characterConfig.characterType;
+  if (characterType) {
+    charType = CharInfo.type.indexOf(characterType);
+    console.log(`Character type has been override, updating - characterType: ${characterType}`);
```

Comment:
This isn't accurate. Since you're defining the characterConfig for this bot in the player code, it will always be defined in the `rg` object - this doesn't necessarily mean that it was overridden during handshake

Reply:
Fixed!